src/go/serviceconfig/rollout_id_change_fetcher_test.go

```diff
@@ -65,22 +66,29 @@ func TestRolloutIdChangeFetcherSetDetectRolloutIdChangeTimer(t *testing.T) {
 	accessToken := func() (string, time.Duration, error) { return "token", time.Duration(60), nil }
 	cif := NewRolloutIdChangeDetector(&http.Client{}, serviceControlServer.GetURL(), "service-name", accessToken)
 
-	cnt := 0
-	wantCnt := 3
+	// `unsafeCnt` will be accessed in muilti-goroutine so please access it using atomic
+	// library.
+	var unsafeCnt, wantCnt uint32
```

Comment:
I am not so sure about this fix.  My understanding is we don't have race

main thread:   cnt = 0,   
timer thread:  t1:  cnt++,   and read,  t2:  cnt++,  and read,   t3:  cnt++,  and read.   and  cnt is not changed since rollout_id is not changed.
main thread:  cnt is read (if it waits long enough for timer thread to pass t3) 

so the sleep at line 88 is important, it should be long enough to pass t3.

I don't see the race.

Reply:
You are totally right. Updated.